cmd/pctl/install.go

```diff
@@ -88,50 +107,68 @@ func installCmd() *cli.Command {
 
 // install runs the install part of the `install` command.
 func install(c *cli.Context) error {
-	profilePath, catalogClient, err := parseArgs(c)
-	if err != nil {
-		_ = cli.ShowCommandHelp(c, "install")
-		return err
+	var (
+		err           error
+		catalogClient *client.Client
+		profilePath   string
+		catalogName   string
+		profileName   string
+		version       = "latest"
+	)
+
+	// only set up the catalog if a url is not provided
+	url := c.String("profile-url")
+	if url != "" && c.Args().Len() > 0 {
+		return errors.New("it looks like you provided a url with a catalog entry; please choose either format: url/branch/path or <CATALOG>/<PROFILE>[/<VERSION>]")
 	}
 
-	branch := c.String("branch")
+	if url == "" {
+		profilePath, catalogClient, err = parseArgs(c)
+		if err != nil {
+			_ = cli.ShowCommandHelp(c, "install")
+			return err
+		}
+		parts := strings.Split(profilePath, "/")
+		if len(parts) < 2 {
+			_ = cli.ShowCommandHelp(c, "install")
+			return errors.New("both catalog name and profile name must be provided")
+		}
+		if len(parts) == 3 {
+			version = parts[2]
+		}
+		catalogName, profileName = parts[0], parts[1]
+	}
+
+	branch := c.String("profile-branch")
 	subName := c.String("subscription-name")
 	namespace := c.String("namespace")
 	configValues := c.String("config-secret")
 	dir := c.String("out")
-
-	parts := strings.Split(profilePath, "/")
-	if len(parts) < 2 {
-		_ = cli.ShowCommandHelp(c, "install")
-		return errors.New("both catalog name and profile name must be provided")
-	}
-	catalogName, profileName := parts[0], parts[1]
+	path := c.String("profile-path")
 
 	fmt.Printf("generating subscription for profile %s/%s:\n\n", catalogName, profileName)
-	version := "latest"
-	if len(parts) == 3 {
-		version = parts[2]
-	}
 	cfg := catalog.InstallConfig{
-		Branch:        branch,
+		ProfileBranch: branch,
 		CatalogName:   catalogName,
 		CatalogClient: catalogClient,
 		ConfigMap:     configValues,
 		Namespace:     namespace,
 		ProfileName:   profileName,
 		SubName:       subName,
 		Directory:     dir,
+		URL:           url,
 		Version:       version,
+		Path:          path,
 	}
 	return catalog.Install(cfg)
 }
```

Comment:
might do this in a follow up PR, this struct is pretty big. Could maybe be split into:
```
InstallConfig{
  CatalogReference
  ProfileReference
  SubscriptionSpec
}
```